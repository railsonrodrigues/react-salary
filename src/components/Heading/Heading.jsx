import React from 'react';
import styled from 'styled-components';

const Title = styled.h1`
  font-size: 48px;
  text-align: ${props => props.align};
`;

const Heading = ({title, align}) => {
  return <Title align={align}>{title}</Title>
}

export default Heading;
