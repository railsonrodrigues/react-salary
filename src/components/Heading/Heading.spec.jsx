import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import React from 'react';
import Heading from './Heading';

describe('<Heading />', () => {
  it('should render heading with text', () => {
    render(<Heading title="Hello World!" align="center" />)
    const heading = screen.getByText('Hello World!');
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent('Hello World!');
  })
  it('should match snapshot', () => {
    const {container} = render(<Heading title="Hello World!" align="center" />)
    expect(container.firstChild).toMatchSnapshot();
  })
})