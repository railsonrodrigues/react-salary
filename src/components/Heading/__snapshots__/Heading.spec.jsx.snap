// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
<h1
  class="sc-bdnxRM kPiEsP"
>
  Hello World!
</h1>
`;
